go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1106,6 +1106,14 @@ func (call *builtinMaketime) compile(c *compiler) (ctype, error) {
 			c.asm.Convert_xf(1)
 			c.asm.Fn_MAKETIME_f()
 		}
+	case sqltypes.Datetime:
+		c.asm.Fn_MAKETIME_D()
+	case sqltypes.Date:
+		c.asm.Fn_MAKETIME_D()
+	case sqltypes.Time:
+		c.asm.Fn_MAKETIME_D()
+	case sqltypes.Timestamp:
+		c.asm.Fn_MAKETIME_D()
```

Comment:
This can be folded into one `case` with multiple types :smile:.

Reply:
😅